exercise/views.py

```diff
@@ -93,6 +95,16 @@ def get(self, request, *args, **kwargs):
                 page.content = _('Unfortunately this exercise is currently '
                                  'under maintenance.')
                 return super().get(request, *args, page=page, students=students, **kwargs)
+        elif self.exercise.status in \
+                (LearningObject.STATUS.ENROLLMENT, LearningObject.STATUS.ENROLLMENT_EXTERNAL):
+            # Retrieve the data of the user's latest enrollment exercise submissions
+            # in case the data could be reused in this enrollment exercise.
+            all_enroll_data = Submission.objects.get_combined_enrollment_submission_data(self.request.user)
+            # Find the intersection of form field keys that are used by the current
+            # exercise and all_enroll_data.
+            # Filter all_enroll_data to only contain these keys.
+            all_enroll_data = filter_dict_with_keys(all_enroll_data,
+                self.exercise.get_form_spec_keys().intersection(all_enroll_data.keys()))
```

Comment:
The intersection could be calculate inside filter_dict_with_keys. Alternatively, I don't see need for the helper, as you could do

```
include_fields = self.exercise.get_form_spec_keys().intersection(all_enroll_data.keys())
all_enroll_data = {key, value for key, value in all_enroll_data.items() if key in include_fields}
```

Reply:
I removed the helper and used this one-liner dict compehension.